test/In.ProjectEKA.HipServiceTest/Link/LinkControllerTest.cs

```diff
@@ -0,0 +1,71 @@
+namespace In.ProjectEKA.HipServiceTest.Link
+{
+    using FluentAssertions;
+    using Hangfire;
+    using Hangfire.Common;
+    using Hangfire.States;
+    using HipLibrary.Patient.Model;
+    using HipService.Discovery;
+    using HipService.Gateway;
+    using HipService.Link;
+    using Microsoft.AspNetCore.Http;
+    using Moq;
+    using Xunit;
+    using static Builder.TestBuilders;
+
+    [Collection("Link Controller Tests")]
+    public class LinkControllerTest
+    {
+        private readonly Mock<LinkPatient> link;
+        private readonly Mock<IDiscoveryRequestRepository> discoveryRequestRepository;
+        private readonly LinkController linkController;
+        private readonly Mock<IBackgroundJobClient> backgroundJobClient;
+
+        public LinkControllerTest()
+        {
+            link = new Mock<LinkPatient>(MockBehavior.Strict, null, null, null, null, null, null);
+            discoveryRequestRepository = new Mock<IDiscoveryRequestRepository>();
+            var gatewayClient = new Mock<GatewayClient>(MockBehavior.Strict, null, null, null);
+
+            backgroundJobClient = new Mock<IBackgroundJobClient>();
+            linkController = new LinkController(
+                discoveryRequestRepository.Object,
+                backgroundJobClient.Object,
+                link.Object,
+                gatewayClient.Object);
+        }
+
+        [Fact]
+        private void ShouldEnqueueLinkRequestAndReturnAccepted()
+        {
+            var faker = Faker();
+            const string programRefNo = "129";
+            const string patientReference = "4";
+            var consentManagerUserId = faker.Random.Hash();
+            var transactionId = faker.Random.Hash();
+            var linkRequest = new PatientLinkReferenceRequest(
+                transactionId,
+                new LinkReference(
+                    consentManagerUserId,
```

Comment:
this field needs to be just "id", please check the API. Also, maybe better to introduce a model "PatientLinkReference" , so that you can support the parallel APIs for the timebeing. If you change the LinkReference.ConsentManagerUserId, then older API may not work anymore. I would mark older LinkReference as Obsolete/deprecated

Reply:
Unless, in C# you can something like "JsonAlias" (alongside jsonProperty). @ganesanarun may have better ideas

Reply:
Fixed